x-pack/plugins/observability_solution/observability_ai_assistant_app/server/functions/query/index.ts

```diff
@@ -132,9 +132,10 @@ export function registerQueryFunction({
         ),
         logger: resources.logger,
         tools: Object.fromEntries(
-          actions
-            .concat(esqlFunctions)
-            .map((fn) => [fn.name, { description: fn.description, schema: fn.parameters }])
+          [...actions, ...esqlFunctions].map((fn) => [
+            fn.name,
+            { description: fn.description, schema: fn.parameters } as ToolDefinition,
```

Comment:
@pgayvallet @arturoliduena `actions.parameters` is typed as `ToolSchema` whereas `esqlFunctions` is typed as `CompatibleJSONSchema`. This gives some incompatibilities when merging the types so I had to do type assertion.

Is the plan to get rid of `CompatibleJSONSchema` in favor of `ToolSchema`?

Reply:
`CompatibleJSONSchema ` is defined [here](https://github.com/sorenlouv/kibana/blob/a39c06452e069f49bed1bf31d12c48698de2850a/x-pack/plugins/observability_solution/observability_ai_assistant/common/functions/types.ts#L38-L46)

Reply:
> Is the plan to get rid of CompatibleJSONSchema in favor of ToolSchema?

I *think* that question is on your side? `CompatibleJSONSchema` is a o11y assistant type, `ToolSchema` is an `inference` plugin's type. You're more than welcome, and encouraged, to drop your own types in favor of the inference framework's, but that's not really my call, is it?

Reply:
Yes, in that case it's mostly a question for @arturoliduena

Reply:
I’m also in favor of moving towards using the `inference` plugin’s types when possible,as it supports a more consistent approach across plugins. However, I had a similar conversation with @dgieselaar and at the moment we agree on keeping o11y assistant types:

https://github.com/elastic/kibana/pull/199286#discussion_r1835447480

> Ideally we encapsulate the event type from the inference client here, and convert it into ChatCompletionChunkEvent. We should eventually completely drop ChatCompletionChunkEvent in favor of the inference plugin's types, but that means that we need to make changes in many places, and there's some stuff that isn't supported yet by the Inference plugin's types. Until that happens, I think we should limit the blast radius and stick to our own types where we can - which should be everywhere except for the implementation of this function, I think.